one_fm/api/doc_events.py

```diff
@@ -40,16 +39,14 @@ def employee_checkin_validate(doc, method):
 			existing_perm = None
 			checkin_time = get_datetime(doc.time)
 			curr_shift = get_current_shift(doc.employee)
-			last_shift = frappe.get_list("Shift Assignment",fields=["*"],filters={"employee":doc.employee},order_by='creation desc',limit_page_length=1)
-			if curr_shift and last_shift:
+			if curr_shift:
 				start_date = (curr_shift.start_date).strftime("%Y-%m-%d")
 				existing_perm = frappe.db.exists("Shift Permission", {"date": start_date, "employee": doc.employee, "permission_type": perm_map[doc.log_type], "workflow_state": "Approved"})
-				doc.shift_assignment = last_shift[0].name

```

Comment:
What business has shift assignment in this story?

Reply:
I'm not sure how it happened but It seems other commits are also added. I did not make those changes in my branch.

Reply:
@Talleyrand333 keep your branches upto date.

Reply:
@mymi14s that seems to be the issue, I did not fetch the latest master-14 branch. I will do that.